### ISSUE TITLE
Installation Issue in clean Ubuntu 22.04 LTS and OMNeT 6.1.0 (WITH_SCAVE_PYTHON_BINDINGS=yes)

### ISSUE THREAD

**Issue by johnuskglass** [omnetpp/omnetpp#1239]
**Platform**
 - OMNeT++ version:6.1.0
 - OS: Ubuntu 22.04.1

**Describe the bug**
I set up a clean Ubuntu 22.04.1 and follow the installation guide as well as running the script install.sh. Sadly, `./configure` always fails because WITH_SCAVE_PYTHON_BINDINGS=yes. Here is my output:

>configure: Environment variables (PATH and PYTHONPATH) are correctly set.
configure: Reading configure.user for your custom settings.
configure: Creating a backup of 'Makefile.inc'.
mv: cannot stat './Makefile.inc': No such file or directory
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for clang... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether clang accepts -g... yes
checking for clang option to enable C11 features... none needed
checking for clang++... clang++
checking whether the compiler supports GNU C++... yes
checking whether clang++ accepts -g... yes
checking for clang++ option to enable C++11 features... unsupported
checking for clang++ option to enable C++98 features... unsupported
checking for clang++... clang++
checking for c++17 support... yes (forced by -std=c++17)
checking for ranlib... ranlib
checking whether LLD linker is available... no
checking whether clang++ supports -fno-omit-frame-pointer... no
checking whether clang++ supports -glldb... no
checking whether clang++ supports -ggdb3... no
checking whether clang++ supports -fstandalone-debug... no
checking whether clang++ supports -fno-limit-debug-info... no
checking whether clang++ supports -Wl,--no-as-needed... no
checking whether clang++ supports -Wl,--as-needed... no
checking for swapcontext... no
checking if shared libs need -fPIC... failure
configure: WARNING: COULD NOT FIGURE OUT HOW TO BUILD SHARED LIBRARY -- reverting to static libs
checking for dlopen with CFLAGS="" LIBS=""... no
checking if --export-dynamic linker option is supported/needed... test failed
checking for flags needed to link with static libs containing simple modules... no solution found
configure: WARNING: Avoid linking with static libraries containing simple modules
configure: WARNING: and other dynamically registered components. Those components
configure: WARNING: will not be found by the simulation program, because the linker
configure: WARNING: leaves them out from the executable.
checking whether linker supports -rpath... no
checking for pkg-config... pkg-config
checking for bison... bison -y
checking for make... make
checking for perl... perl
checking for swig... not found
checking for python3... python3
checking if all necessary Python modules are available... yes
checking for python3-config... /usr/bin/python3-config
checking for python3-dev with CFLAGS="-I/usr/include/python3.10 -I/usr/include/python3.10" LIBS="-L/usr/lib/python3.10/config-3.10-x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu -lpython3.10 -lcrypt -ldl  -lm -lm "... no
configure: error: Python support was configured (WITH_SCAVE_PYTHON_BINDINGS=yes), but the Python 3 development package was not found.

**To Reproduce**
Create a clean Ubuntu 22.04.1 and try to install OMNeT 6.1.0 with ./configure.

**Comment by johnuskglass** [comment]
Fixed it by running `sudo apt install libstdc++-12-dev`

**Comment by rhornig** [comment]
Interesting because `libpython3-dev` is explicitly installed by the install script. Does the same happen with the ubuntu 24.04 image?

**Comment by rhornig** [comment]
This is an unrelated issue caused by missing the `python3-config` command. What python are you using. Did you set up the environment from HOMEBREW and the executed the instal.sh as suggested by the install instruction?

**Comment by Cool-baby** [comment]
Sorry, I looked at the Installation Guide PDF file before and found that the MACOS system mentioned starting with `./configure`. So, I followed this file step by step and encountered the problem I encountered earlier. After checking the GitHub source code address, I saw that the `install. sh` script could be used to install directly, and I successfully installed it using this script. Thank you. Additionally, the Installation Guide PDF file may need to be updated regarding the installation method of the `install. sh` script.

**Comment by rhornig** [comment]
Indeed, we will update the install guide, but `install.sh` just executes what is written down in the guide. You probably missed this part and that caused the issue:

> Restart your shell to activate the above changes and make sure that you are using python3
> from Homebrew (which python3) and not the system’s Python interpreter provided by macOS
> (/usr/lib/python3) and then install the Python dependencies:

You probably started configure while the python interpreter was still the system one. Anyway, I'm happy that the automated script worked.

**Comment by rhornig** [comment]
Ok, long story short, there is an error in the dependencies (on debian/ubuntu). You need `python3-dev` instead of `libpython3-dev` because the later does not contain the python3-config tool.

**Comment by Ricky12121228** [comment]
> Fixed it by running `sudo apt install libstdc++-12-dev`
really helpful  thx u

**Comment by rhornig** [comment]
Indeed, the python binding was using static linking with libstdc++, and the static lib dev package was not documented as a dependency. Using static linking is a bad idea anyway, so we switched to dynamic linking. See efe7c1db324a3bb32ed52755f32141573fe5f7cb
